### ISSUE TITLE
[Enhancement]: Enable New Feature: Stage Level Rollback

### ISSUE THREAD

**Issue by mw-root** [hashicorp/terraform-provider-aws#37244]
### Description

CodePipeline recently announced an option for stage-level rollbacks.

Announcement: https://aws.amazon.com/about-aws/whats-new/2024/04/aws-codepipeline-stage-level-manual-automated-rollback/
Configuration: https://docs.aws.amazon.com/codepipeline/latest/userguide/stage-rollback.html

### Affected Resource(s) and/or Data Source(s)

aws_codepipeline

### Potential Terraform Configuration

```terraform
resource "aws_codepipeline" "pipeline" {
  name = "my-pipeline"
  ...

  stage {
    name       = "Deploy"
    on_failure = "ROLLBACK"
    ...
  }
}
```

### Would you like to implement a fix?

Yes

PR: #37245

**Comment by novekm** [comment]
+1 this would be very helpful

**Comment by mw-root** [comment]
Could we get some eyes on this?

The associated PR is tiny and it really would be great to have this functionality.

**Comment by duncangammie-addepar** [comment]
What's the status on this? We would really like this feature in our work.

**Comment by mw-root** [comment]
If anyone has ideas for getting this looked at, I'm all ears. 

The PR is nearly as simple as could be and I am sure once someone looks at it that it will be a quick win.

**Comment by duncangammie-addepar** [comment]
> If anyone has ideas for getting this looked at, I'm all ears.
> 
> The PR is nearly as simple as could be and I am sure once someone looks at it that it will be a quick win.

Yep. Instead, our team is implementing a Github Action that applies the CLI command to update the pipelines to rollback on failure, which is sort of a hacky/annoying solution.

**Comment by mw-root** [comment]
This issue and PR were open for almost a year. 

Someone duplicates the work and gets it merged in a few months time. 

What part of the secret handshake am I missing here?
